Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -550,6 +557,8 @@ final class PurchasesOrchestrator {
 
         if let transaction = transaction {
             customerInfo = try await self.handlePurchasedTransaction(transaction, .purchase)
+
+            self.postPaywallEventsIfNeeded()
```

Comment:
Again I think this is called after a successful SK2 purchase. I think every SK2 purchase made using the SDK goes through this path, but let me know if there's somewhere else I missed.

Reply:
I think the only other place would be purchases that happen outside of the app, or basically other purchases in the queue.

I think the safest would be to sync in the same places we are syncing `unsyncedAttributes`. If you look at `storeKit2TransactionListener`, we add the unsynced attributes to `PurchasedTransactionData`, and post everything together. I think every single purchase goes through `storeKit2TransactionListener`

Since these are paywall events, I think those places are not that important. Something I was thinking is what happens if the transaction fails posting in a paywall? I think the event will be posted next time the app is opened and it should be safe, but maybe it's safer to just try to post events in `storeKit2TransactionListener`?

Reply:
I wanted to keep posting to only "newly purchased transactions via a paywall". If we sync in the listener, we'll be doing it for subscription updates, and purchases made outside the app which never generate paywall events... and yes those will be synced anyway when the app is backgrounded/foregrounded.